apps/backend/src/bookings/bookings.controller.ts

```diff
@@ -21,21 +53,157 @@ export class BookingsController {
     private readonly bookingsService: IBookingsService,
   ) {}
 
-  @Get()
-  async getBookings() {
-    // Todo: Implement this
+  @ApiCookieAuth('access_token')
+  @ApiOperation({
+    summary: 'get bookings by user ID in the session',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'successfully get the bookings',
+    type: GetBookingsByUserIdResponse,
+  })
+  @ApiResponse({
+    status: HttpStatus.UNAUTHORIZED,
+    description: 'valid session is not provided',
+  })
+  @ApiResponse({
+    status: HttpStatus.INTERNAL_SERVER_ERROR,
+    description: 'internal server error',
+  })
+  @UseGuards(AuthGuard)
+  @Get('/self')
+  async getBookings(@Req() req, @Res({ passthrough: true }) res) {
+    try {
+      const account: AccountMetadata = req.account;
+      const resBody = await this.bookingsService.getBookingsByUserId({
+        userId: account.userId,
+      });
+      res.status(HttpStatus.OK).send(resBody);
+    } catch (e) {
+      console.log(e);
+      if (e instanceof RecordNotFound) {
+        throw new HttpException({ message: e.message }, HttpStatus.NOT_FOUND);
```

Comment:
[Trivia] should respond 500 instead since "userId" from AccountMetadata must exists.